domain/unitstate/state/state.go

```diff
@@ -29,31 +30,24 @@ func NewState(factory database.TxnRunnerFactory) *State {
 
 // GetUnitUUIDForName returns the UUID corresponding to the input unit name.
 // If no unit with the name exists, a [errors.UnitNotFound] error is returned.
-func (st *State) GetUnitUUIDForName(ctx domain.AtomicContext, name string) (string, error) {
-	uName := unitName{Name: name}
-	uuid := unitUUID{}
-
-	q := "SELECT &unitUUID.uuid FROM unit WHERE name = $unitName.name"
-	stmt, err := st.Prepare(q, uName, uuid)
-	if err != nil {
-		return "", errors.Errorf("preparing UUID query: %w", err)
-	}
-
-	err = domain.Run(ctx, func(ctx context.Context, tx *sqlair.TX) error {
-		err := tx.Query(ctx, stmt, uName).Get(&uuid)
-		if errors.Is(err, sqlair.ErrNoRows) {
-			return uniterrors.UnitNotFound
-		}
+func (st *State) GetUnitUUIDForName(ctx domain.AtomicContext, name coreunit.Name) (coreunit.UUID, error) {
+	var unitUUID unitUUID
+	err := domain.Run(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		var err error
+		unitUUID, err = st.getUnitUUIDForName(ctx, tx, name)
 		return err
 	})
```

Comment:
what's the point of keeping the run atomic here? it's easier to do:
```
err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
        unitUUID, err = st.getUnitUUIDForName(ctx, tx, name)
        return err
})
```
right?

Reply:
discussed offline

Reply:
Because the outer service method calls `GetUnitUUIDForName` in an atomic context.

See:
```
// SetState persists the input unit state selectively,
// based on its populated values.
func (s *Service) SetState(ctx context.Context, as unitstate.UnitState) error {
	return s.st.RunAtomic(ctx, func(ctx domain.AtomicContext) error {
		uuid, err := s.st.GetUnitUUIDForName(ctx, as.Name)
		if err != nil {
			return errors.Errorf("getting unit UUID for %s: %w", as.Name, err)
		}

		if err = s.st.EnsureUnitStateRecord(ctx, uuid); err != nil {
			return errors.Errorf("ensuring state record for %s: %w", as.Name, err)
		}
...
```